### PR TITLE
Trigger error when property is removed

### PR DESCRIPTION
**Pull request by ismail1432**
The goal of the PR is to trigger an error when a property is removed. 

To assert if a property was removed we compare the properties of the message version N+1 to the properties of the message version N.

The PR introduce  2 new DTO (`Property` ans `PropertyList` ) for handling the properties of the message with an object API.

### CODE REVIEW

File: src/Factory/ZddPropertyExtractor.php [hunk 1]
```diff
@@ -37,8 +37,6 @@ public function extractPropertiesFromClass(string $className): PropertyList
             }
 
             if ($propertyType->allowsNull()) {
-                $propertyList->addNullableProperty($propertyName);
```
Comment (Inkod) [inline src/Factory/ZddPropertyExtractor.php:40]: If I understand correctly, we should remove this condition completly no ?

Comment (ismail1432) [reply src/Factory/ZddPropertyExtractor.php:40]: Yes we can remove it :-)

Comment (Inkod) [reply src/Factory/ZddPropertyExtractor.php:40]: After discussion, we need to keep it and add property with null value to be sure the property will not be set to not null

**Review by Inkod — APPROVED** [review]
Great job ! :tada:

**Pull request merged**